src/ui/component-repository/components/list/HubRegistrySwitcher.js

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useEffect } from 'react';
+import { Row, Col, DropdownKebab, MenuItem, Icon } from 'patternfly-react';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+
+import { fetchRegistries, setActiveRegistry, setFetchedBundleGroups, setFetchedBundlesFromRegistry } from 'state/component-repository/hub/actions';
+import { getRegistries, getSelectedRegistry } from 'state/component-repository/hub/selectors';
+import { ECR_LOCAL_REGISTRY_NAME } from 'state/component-repository/hub/reducer';
+import { setVisibleModal, setInfo } from 'state/modal/actions';
+import { ADD_NEW_REGISTRY_MODAL_ID } from 'ui/component-repository/components/list/AddNewRegistryModal';
+import { DELETE_REGISTRY_MODAL_ID } from 'ui/component-repository/components/list/DeleteRegistryModal';
+
+const DEFAULT_ECR_REGISTRY = {
+  name: ECR_LOCAL_REGISTRY_NAME,
+  url: '',
+};
+
+const HubRegistrySwitcher = () => {
+  const activeRegistry = useSelector(getSelectedRegistry);
+  const registries = useSelector(getRegistries);
+  const dispatch = useDispatch();
+
+  const handleRegistryChange = useCallback((registry) => {
+    if (registry.name !== activeRegistry.name) {
+      dispatch(setFetchedBundleGroups([]));
+      dispatch(setFetchedBundlesFromRegistry([]));
+      dispatch(setActiveRegistry(registry));
+    }
+  }, [activeRegistry.name, dispatch]);
+
+  const handleNewRegistryClick = () => {
+    dispatch(setVisibleModal(ADD_NEW_REGISTRY_MODAL_ID));
+    dispatch(setInfo({ type: 'Registry' }));
+  };
+
+  const handleDeleteRegistry = (registry) => {
+    if (activeRegistry.id === registry.id) return;
+    dispatch(setVisibleModal(DELETE_REGISTRY_MODAL_ID));
+    dispatch(setInfo({ type: 'Registry', code: registry.name, id: registry.id }));
+  };
+
+  useEffect(() => { dispatch(fetchRegistries()); }, [dispatch]);
+
+  useEffect(() => {
+    handleRegistryChange(DEFAULT_ECR_REGISTRY);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why should exhaustive-deps rule be disabled here? Is is strongly recommended to keep it enabled.

Reply:
removed, done ✅

Reply:
I cannot remove this. As I need that effect to run *only* when mounting (empty array)

Reply:
When I am removing the eslint line, it automatically populated with deps array... which I dont want. I want explicitly when a component mounts, no more.